python/tank/deploy/tank_commands/unregister_folders.py

```diff
@@ -0,0 +1,311 @@
+        # linked up to the folders registered for this entity.
+        
+        paths = []
+        pc = path_cache.PathCache(self.tk)
+        try:
+            for sg_fs_id in ids:
+                paths.extend(pc.get_folder_tree_from_sg_id(sg_fs_id))
+        finally:
+            pc.close()
+                
+        log.info("")
+        log.info("The following folders will be unregistered:")
+        for p in paths:
+            log.info(" - %s" % p["path"])
+        
+        log.info("")
+        log.info("Proceeding would unregister the above paths from Toolkit's folder database. "
+                 "This will not alter any of the content in the file system, but once you have "
+                 "unregistered the paths, they will not be recognized by Shotgun.")
+        log.info("")
+        log.info("This is useful if you for example have renamed an Asset or Shot and want to "
+                 "move the data around on disk. In this case, start by unregistering the existing "
+                 "folder entries in Shotgun, then rename the Shot or Asset in Shotgun. Next, "
+                 "create new folders on disk using Toolkit's 'create folders' command and lastly "
+                 "move all your file system data across from the old location to the new location.")
+        log.info("")
+        if prompt:
+            val = raw_input("Proceed with unregister? (Yes/No) ? [Yes]: ")
+            if val != "" and not val.lower().startswith("y"):
+                log.info("Exiting! Nothing was unregistered.")
+                return []
+        
+        log.info("Removing items from Shotgun....")
+        log.info("")
+        
+        sg_batch_data = []
+        for p in paths:                            
+            req = {"request_type":"delete", 
+                   "entity_type": path_cache.SHOTGUN_ENTITY, 
+                   "entity_id": p["sg_id"] }
+            sg_batch_data.append(req)
+        
+        try:    
+            self.tk.shotgun.batch(sg_batch_data)
+        except Exception, e:
+            raise TankError("Shotgun Reported an error. Please contact support. "
+                            "Details: %s Data: %s" % (e, sg_batch_data))
+        
+        # now register the deleted ids in the event log
+        # this will later on be read by the synchronization            
+        # now, based on the entities we just deleted, assemble a metadata chunk that 
+        # the sync calls can use later on.
+        
+        pc_link = {"type": "PipelineConfiguration",
+                   "id": self.tk.pipeline_configuration.get_shotgun_id() }
+        
+        project_link = {"type": "Project", 
+                        "id": self.tk.pipeline_configuration.get_project_id() }
+        
+        meta = {}
+        # the api version used is always useful to know
+        meta["core_api_version"] = self.tk.version
+        # shotgun ids created
+        meta["sg_folder_ids"] = [ x["sg_id"] for x in paths]
```

Comment:
extra (or missing) space depending on the style guide:  `[x["sg_id"] for x in paths]`